src/graph/matrix/delta_matrix_iter.rs

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+use super::{
+    delta_matrix::DeltaMatrix, sparse_matrix_iter::SparseMatrixIter, GraphBLAS::GrB_Index,
+};
+
+pub struct DeltaMatrixIter<'a> {
+    matrix: Option<&'a DeltaMatrix>,
+    m_it: SparseMatrixIter,
+    dp_it: SparseMatrixIter,
+    min_row: GrB_Index,
+    max_row: GrB_Index,
+}
+impl<'a> DeltaMatrixIter<'a> {
```

Comment:
should maybe implement Iterator? if you can think of a next() function which returns an Option<&'a DeltaMatrix>

Reply:
(it will allow us to iterate over all the delta matrices in a Rustic way, then we can use all of Rust's optimized iterator functions(map, filter, fold etc.)